### PR TITLE
Create house_robber.cpp

### PR DESCRIPTION
**Pull request by SumukhBhardwaj**
Added House Robber problem, solved using dynamic programming

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1247"></a>

### CODE REVIEW

**Comment by kvedala** [comment]
Does your code conform to the standards and the checklist?
image

**Comment by SumukhBhardwaj** [comment]
> Does your code conform to the standards and the checklist?
> image

Yes I have added enough comments and also take care of naming conventions.

**Comment by kvedala** [comment]
It does not follow the repo standards. All the details are provided in the links in the checklist. Please review and update the code accordingly. No reviews will be provided until the code is updated to the repo standards. 
Thank you for your contribution.

**Comment by SumukhBhardwaj** [comment]
> It does not follow the repo standards. All the details are provided in the links in the checklist. Please review and update the code accordingly. No reviews will be provided until the code is updated to the repo standards.
> Thank you for your contribution.

Please review now, i have added test cases and solution as well.

**Comment by SumukhBhardwaj** [comment]
@kvedala  please review it now. I have done all the required changes.

File: dynamic_programming/house_robber.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
+/*House Robber - Dynamic Programming
```
Comment (kvedala) [inline dynamic_programming/house_robber.cpp:1]:
```suggestion
/**
 * @file 
 * @brief House Robber - Dynamic Programming
```
Update the documetnation block per the guidelines. A sample code is provided for your convenience. In the above comment, an example existing code was also provided as a reference.

File: dynamic_programming/house_robber.cpp [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+*/
+
+
+#include <iostream>
+#include <vector> 
+#include <cassert>
+
+/**
+ * @brief This function implements the above algorithm
+ * @money array of numbers containing the money in the ith house
```
Comment (kvedala) [inline dynamic_programming/house_robber.cpp:28]: the doc blocks are not per the requested, suggested and described in the guidelines. refer to the comment above.

The "This" does not reflect in the documentation. You can review the documentation site: http://thealgorithms.github.io/C-Plus-Plus
You can split the file and function documentations. See [this file](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/dynamic_programming/kadane2.cpp) for example. Also note the use of namespaces.

**Review by kvedala — CHANGES_REQUESTED** [review]
👍 
Good update but much needs to be done still. Please review the comments thoroughly.

**Comment by Swastyy** [comment]
@SumukhBhardwaj hey! where did you find the problem statement? 
Can you please tell me how can I share my solution?
(P.S. I am new to open source and need some initial help.)

**Pull request closed without merging**